src/dialect/onnx/onnx_ops.cpp

```diff
@@ -781,7 +781,8 @@ void ONNXConvNoBiasOp::inferShapes() {
   int64_t group =
       ONNXConvNoBiasOp::group().getSExtValue(); //.getLimitedValue();
   // Check that the X.shape[1] == (W.shape[1] * group) == C condition holds.
-  if (dataShape[1] != (weightShape[1] * group))
```

Comment:
question: the dataShape array's default value is "-1"?

Reply:
Data is a required input so its values are from a previous op's output. If the data is a tensor containing unknown dimensions (e.g. tensor<32x?x32xf32>), the its dataShape's values should contain `-1` (e.g. dataShape[1] of vector<32x?x32xf32> is `-1`). For unknown dimensions, we do not know whether dimensions mismatched or not at the time of shape inference.

Reply:
Thanks a lot for clarification! I was trying to find/learn the codes used to set the dataShape[I] to be "-1" when its dimension is unknown, otherwise, the int default value would be "0".

Reply:
`-1` is used to represent unknown shape element. You might find this helpful: https://github.com/llvm/llvm-project/blob/master/mlir/include/mlir/IR/StandardTypes.h#L345